src/ring_theory/polynomial/hermite.lean

```diff
@@ -125,4 +129,32 @@ end
 
 end coeff
 
+/-! ### Lemmas about `hermite n` and gaussians -/
+section gaussian
+
+lemma hermite_gauss_succ (n : ℕ) (x : ℝ) :
+  (-1 : ℝ)^(n+1) * real.exp (x^2 / 2) * (deriv^[n+1] (λ y, real.exp (-(y^2 / 2))) x)
+  = x * ((-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)) -
+    deriv (λ x, (-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)) x :=
+begin
+  rw [function.iterate_succ_apply', deriv_mul, deriv_const_mul,
+      (by simp [mul_comm] : deriv (λ y, real.exp (y^2 / 2)) x = x * real.exp (x^2 / 2))],
+  ring_exp,
+  { simp },
+  { simp },
+  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterate_deriv _ _ _)).1,
+    exact ((cont_diff_id.pow 2).div_const 2).neg.exp },
+end
```

Comment:
I'd recommend inlining this result into the proof below. You could even keep the `gaussian` def locally, as
```
  let gaussian : ℝ → ℝ := λ x, real.exp (-(x^2 / 2)),
  have : (-1 : ℝ)^(n+1) * (deriv^[n+1] gaussian x) / gaussian x
    = x * ((-1 : ℝ)^n * (deriv^[n] gaussian x)) - deriv (λ y, (-1 : ℝ)^n * (deriv^[n] gaussian y)) x,
  { sorry },
```

Reply:
I moved it inline but decided not to reintroduce `gaussian` (without the auxiliary lemmas, it's more trouble than it's worth).